build_runner/lib/src/generate/build_impl.dart

```diff
@@ -249,9 +254,15 @@ class BuildImpl {
       // Yield the outputs.
       for (var output in wrappedWriter.assetsWritten) {
         (_assetGraph.get(output) as GeneratedAssetNode).wasOutput = true;
-        yield output;
+        streamController.add(output);
       }
     }
+
+    Future.wait(primaryInputs.map(runForInput)).then((_) {
```

Comment:
[nit] should we change the return type to `Future<Iterable>` instead of using a `StreamController`?

Reply:
Sure it doesn't seem like we are taking advantage of the stream anyways, I can follow up